forest/sycamore/survey_config.py

```diff
@@ -99,121 +142,224 @@ def gen_survey_schedule(config_path, time_start, time_end, beiwe_ids):
+            schedule
+        all_interventions_dict(dict):
+            dict containing interventions for each user. Each key in the dict
+            is a beiwe id. Each value is a dict, with a key for each
+            intervention name and a timestamp for each intervention's time
 
     Returns:
-        A DataFrame with all surveys deployed in the given timeframe on the study to the users with completion times
+        submit_lines(DataFrame): A DataFrame with all surveys deployed in the
+        given timeframe on the study to the users with completion times
+        submit_lines_summary(DataFrame): a DataFrame with all users, total
+        surveys received, and responses.
     """
-    time_start = pd.Timestamp(time_start)
-    time_end = pd.Timestamp(time_end)
-    # Generate aggregated survey data
-    #     agg = functions.aggregate_surveys_config(study_dir, config_path, study_tz)
-
-    # Generate scheduled surveys data
-    sched = gen_survey_schedule(config_path, time_start, time_end, beiwe_ids)
-
-    # Merge survey submit lines onto the schedule data and identify submitted lines
+    if all_interventions_dict is None:
+        all_interventions_dict = {}
+
+    sched = gen_survey_schedule(
+        config_path,
+        time_start,
+        time_end,
+        beiwe_ids,
+        all_interventions_dict)
+
+    if sched.shape[0] == 0:  # return empty dataframes
+        print("Error: No survey schedules found")
+        return (pd.DataFrame(columns=[['survey id', 'beiwe_id']]),
+                pd.DataFrame(columns=[['survey id', 'beiwe_id']]))
+
+    # Merge survey submit lines onto the schedule data and identify submitted
+    # lines
     submit_lines = pd.merge(
-        sched[['delivery_time', 'next_delivery_time', 'id', 'beiwe_id']].drop_duplicates(),
-        agg[['Local time', 'config_id', 'survey id', 'beiwe_id']].loc[agg.submit_line == 1].drop_duplicates(),
+        sched[['delivery_time', 'next_delivery_time',
+               'id', 'beiwe_id']].drop_duplicates(),
+        agg[['Local time', 'config_id', 'survey id', 'beiwe_id']
+            ].loc[agg.submit_line == 1].drop_duplicates(),
         how='left',
         left_on=['id', 'beiwe_id'],
         right_on=['config_id', 'beiwe_id'])
 
-    # Get the submigged survey line
+    # Get the submitted survey line
     submit_lines['submit_flg'] = np.where(
         (submit_lines['Local time'] >= submit_lines['delivery_time']) &
         (submit_lines['Local time'] < submit_lines['next_delivery_time']),
         1, 0
     )
 
     # Take the maximum survey submit line
-    submit_lines2 = submit_lines.groupby(['delivery_time', 'next_delivery_time', 'survey id', 'beiwe_id', 'config_id'])[
-        'submit_flg'].max().reset_index()
+    submit_lines2 = submit_lines.groupby(
+        [
+            'delivery_time',
+            'next_delivery_time',
+            'survey id',
+            'beiwe_id',
+            'config_id'
+        ])['submit_flg'].max().reset_index()
```

Comment:
`)['submit_flg'].max().reset_index()` should be on a new line to match the opening parenthesis

Reply:
This has been updated